container-core/src/test/java/com/yahoo/container/di/ContainerTest.java

```diff
@@ -282,8 +282,9 @@ void getNewComponentGraph_hangs_waiting_for_valid_config_after_invalid_config()
                      () -> getNewComponentGraph(container, currentGraph));
 
         ExecutorService exec = Executors.newFixedThreadPool(1);
+        dirConfigSource.clearCheckedConfigs();
         Future<ComponentGraph> newGraph = exec.submit(() -> getNewComponentGraph(container, currentGraph));
-
+        dirConfigSource.awaitConfigChecked(10_000);
```

Comment:
Without this, the reload below can be partially eaten by the other thread, resulting in a state where it fails the first run through all configs because of different generations, and never completes the second run because some configs don't have anything new. 
That this is possible doesn't seem right at all. I hope it's not the case for the actual JRT subscription, i.e., that that one ensures atomicity in some way, to avoid this state?